File: Leetcode/Length of Last Word/Length_of_last_word.cpp
// * ALGORITHM =>
// * As we have to find the length of last word, we can start at the end of the given string and traverse the string backwards till we encounter first space.
// * Although there are some edge cases which needs to be taken care of. Such as for "a" and " a" and "a ".
// * So will maintain a count variable initialised to '0' and only increment count variable on encountering a character which is not a space.
// * Also, we will only break out of the loop when either we reach the start of the string (i < 0) or we encounter a space and the count is not equal to 0, to handle the edge case of "a "


// * Driver code + lengthofLastWord function
class Solution{
public:
    int lengthOfLastWord(string s){
        int count = 0;
        for (int i = s.size() - 1; i >= 0; i--){
            if (s[i] == ' ' && count != 0){
                break;
            }
            if (s[i] != ' '){
                count++;
            }
        }
        return count;
    }
};

File: Leetcode/Length of Last Word/LengthOfLastWord.py

# Algorithm: 
# 	The approach is to find the index one lesser than starting index, and the ending index of the last word
#   Therefore, we traverse the string from the end
#   Once the indices are found, we subtract it, if it is lesser than 0, it means there is no last word, so we return zero
#   Else, we return the length of the last word

class Solution:
	# Length Of Last Word function
    def lengthOfLastWord(self, s: str) -> int:

    	# 'Pointer - 1', to find the ending index of the last word, initialised to last index of the word
        i = len(s) - 1

        # Moving the 'pointer - 1' towards left, until a non-space character is found or it reached an invalid value 
        while(i >= 0 and s[i] == ' '):
            i -= 1

        # 'Pointer - 2', to find the index one lesser than the starting index of the last word, initialised with the ending index of the last word
        j = i

        # Moving the 'pointer - 2' towards left, until a space is found or it reached an invlid value
        while(j >= 0 and s[j] != ' '):
            j -= 1

        # If the subraction of the indices gives value less than zero, return zero, else return the length of the last word
        length = i - j
        if(length < 0):
            length = 0
        return length

        